test/hotspot/jtreg/compiler/c2/irTests/TestShiftAndMask.java

```diff
@@ -455,30 +566,30 @@ public static void shiftConvMaskCheckIndex_runner() {
     @IR(counts = { IRNode.AND_L, "1" })
     @IR(failOn = { IRNode.ADD_L, IRNode.CONV_I2L })
     public static long addShiftConvMaskCheckIndex(int i, long j, int length) {
-        return (j + Objects.checkIndex(i << 2, length)) & 3; // transformed to: return j & 3;
+        return (j + Objects.checkIndex(i << INT_MASK_WIDTH, length)) & INT_MASK; // transformed to: return j & INT_MASK;
     }
 
     @Run(test = "addShiftConvMaskCheckIndex")
     public static void addShiftConvMaskCheckIndex_runner() {
-        int i = RANDOM.nextInt((Integer.MAX_VALUE - 1) >> 2);
+        int i = RANDOM.nextInt((Integer.MAX_VALUE - 1) >> INT_MASK_WIDTH);
         long j = RANDOM.nextLong();
-        long res = addShiftConvMaskCheckIndex(i, j, (i << 2) + 1);
-        if (res != (j & 3)) {
+        long res = addShiftConvMaskCheckIndex(i, j, (i << INT_MASK_WIDTH) + 1);
+        if (res != (j & INT_MASK)) {
             throw new RuntimeException("incorrect result: " + res);
         }
     }
 
     @Test
     @IR(failOn = { IRNode.AND_L, IRNode.ADD_L })
     public static long addShiftConvMaskCheckIndex2(int i, int j, int length) {
-        return (((long)Objects.checkIndex(j << 2, length)) + ((long)Objects.checkIndex(i << 2, length))) & 3; // transformed to: return 0;
+        return (((long) Objects.checkIndex(j << INT_MASK_WIDTH, length)) + ((long) Objects.checkIndex(i << INT_MASK_WIDTH, length))) & INT_MASK; // transformed to: return 0;
     }
 
     @Run(test = "addShiftConvMaskCheckIndex2")
     public static void addShiftConvMaskCheckIndex2_runner() {
-        int i = RANDOM.nextInt((Integer.MAX_VALUE - 1) >> 2);
-        int j = RANDOM.nextInt((Integer.MAX_VALUE - 1) >> 2);
-        long res = addShiftConvMaskCheckIndex2(i, j, (Integer.max(i, j) << 2) + 1);
+        int i = RANDOM.nextInt((Integer.MAX_VALUE - 1) >> INT_MASK_WIDTH);
+        int j = RANDOM.nextInt((Integer.MAX_VALUE - 1) >> INT_MASK_WIDTH);
+        long res = addShiftConvMaskCheckIndex2(i, j, (Integer.max(i, j) << INT_MASK_WIDTH) + 1);
```

Comment:
I wonder if it would make sense to keep the old tests, and just add new ones with random constants. That way if we mess up with the new changes, we would at least still have the old ones catching us.

Reply:
I'll let you decide - but I do find the combinatorial of {int, long, i2l} x {const, nonconst mask} x {left, right} x {+shift, +const} pretty unwieldy already, duplicating the tests once more would just be piling on.

I could move the randomized tests to a new file name, keep the old as is, and once you have sufficient confidence, you delete the old ones? I don't see they can be maintained in parallel for long.